cloud/pkg/dynamiccontroller/application/application.go

```diff
@@ -313,14 +315,19 @@ func (a *Agent) Apply(app *Application) error {
 	case Completed:
 		app.Reset()
 		go a.doApply(app)
-	case Rejected, Failed:
+	case Rejected:
+		return &app.Error
+	case Failed:
 		return errors.New(app.Reason)
 	case Approved:
 		return nil
 	case InApplying:
 		//continue
 	}
 	app.Wait()
+	if app.getStatus() == Rejected {
+		return &app.Error
+	}
```

Comment:
Why repeating for `Rejected` here?

Reply:
> Why repeating for `Rejected` here?

wait for applying status finish here, if app response status is Rejected, return k8s error directly.